Allow composite key

ataft opened the pull request:
Fix migration to use idNames() instead of idName(), allowing for a composite primary key from multiple columns.

If you look at migration.js [line 392](https://github.com/loopbackio/loopback-connector-mssql/blob/master/lib/migration.js#L392), you can see it only adds one column to the primary key. This is because migration.js [line 358](https://github.com/loopbackio/loopback-connector-mssql/blob/master/lib/migration.js#L358) calls `this.idName(model)` instead of `this.idNames(model)`.

Fixes https://github.com/loopbackio/loopback-connector-mssql/issues/1 and https://github.com/loopbackio/loopback-connector-mssql/issues/138

dhmlau commented:
@ataft, thanks for your contribution. Could you please add some tests to verify your changes? Thanks.

ataft commented:
Test has been added

dhmlau commented:
@ataft, thanks for adding the test. Your changes look reasonable to me. Since CI is not working and I have problems in getting my mssql docker image running for local tests, I'd prefer not to approve the PR until then. 
And I'd also like to get at least one more review. 

In the meanwhile, I'll see if I can get the GitHub Actions working or the docker image running.

marioestradarosa commented:
@ataft please squash your commits. Also the issue #1 is not closed by this PR, which is trying to prevent the DB server to auto generate an id by using the flag `idInjection`. The fact that a work around is to use a composite key, not all use cases are using a composite one, so please update your PR description. 

Also, it is important to note that `loopback4` does not support composite keys when performing CRUD operations, so this PR only addresses the migration part.

ataft commented:
I squashed the commits and removed the text about closing issue #1.

Regarding Loopback 4 not supporting composite keys, that's fine since we'll be on Loopback 3 until v4 comes closer to reaching parity with v3. I was keeping track of the features in v3 that were not in v4, but it got a little out of control, as you can see in the bot-closed issue: https://github.com/loopbackio/loopback-next/issues/1920.

dhmlau reviewed (approved):
The changes look reasonable to me. I also was able to run the tests locally. Would like to get at least another approval (possibly from @marioestradarosa) before landing this PR. Thanks.

dhmlau commented:
FYI - I've updated the setup.sh script to set up the docker image for sql server: https://github.com/loopbackio/loopback-connector-mssql/pull/237.
